include/sys/multi_heap.h

```diff
@@ -125,6 +132,17 @@ void *sys_multi_heap_alloc(struct sys_multi_heap *mheap, void *cfg, size_t bytes
 void *sys_multi_heap_aligned_alloc(struct sys_multi_heap *mheap,
 				   void *cfg, size_t align, size_t bytes);
 
+/**
+ * @brief Get a specific heap for provided address
+ *
+ * Returns the pointer to a single system heap (with metadata)
+ *  controlling the provided pointer
```

Comment:
Pedantry: we should probably specify that the pointer must be an un-freed return value from sys_multi_heap_alloc().  It's true that right now, it's just a search of address ranges so it will work for any address inside the heap memory.  But it's easy to imagine a more complicated scheme showing up in the future where there's a header in the block or something.